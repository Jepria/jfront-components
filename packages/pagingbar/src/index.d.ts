import React from 'react';
interface PagingToolBarProps {
    startPageNumber?: number;
    pageCount: number;
    onChange?: (currentPageNumber: number) => void;
}
export declare const PagingToolBar: React.FC<PagingToolBarProps>;
export {};
