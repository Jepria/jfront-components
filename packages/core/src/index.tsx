export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
