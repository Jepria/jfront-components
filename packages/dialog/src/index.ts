export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'