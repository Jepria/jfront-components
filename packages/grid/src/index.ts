export * from './common'
export * from './jepgrid'
