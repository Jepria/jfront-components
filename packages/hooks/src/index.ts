export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
