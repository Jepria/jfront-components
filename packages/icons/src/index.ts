export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
