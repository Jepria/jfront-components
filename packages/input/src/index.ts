export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
