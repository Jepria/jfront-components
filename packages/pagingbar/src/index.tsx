import React from "react"
import styled from "styled-components"
import {
  FirstImage,
  PrevImage,
  NextImage,
  LastImage,
  RefreshImage,
} from "@jfront/ui-icons"
import { Label } from "@jfront/ui-label"
import { NumberInput } from "@jfront/ui-input"

const Wrapper = styled.div`
  display: inline-flex;
  align-items: center;
  white-space: nowrap;
`

const Item = styled.button`
  height: 100%;
  padding: 1px 1px;
  margin: 0 1px;
  border: ${(props) =>
    `${props.theme.toolbar.button.borderWidth} 
    ${props.theme.toolbar.button.borderStyle} 
    ${props.theme.toolbar.button.borderColor}`};
  background: ${(props) => props.theme.toolbar.button.bgColor};
  color: ${(props) => props.theme.toolbar.button.color};
  &:disabled {
    opacity: 0.5;
    background: transparent;
    cursor: default;
  }
  &:enabled {
    &:hover {
      background: ${(props) => props.theme.toolbar.button.hoverBgColor};
      border: ${(props) =>
        `${props.theme.toolbar.button.hoverBorderWidth} 
        ${props.theme.toolbar.button.hoverBorderStyle} 
        ${props.theme.toolbar.button.hoverBorderColor}`};
    }
  }
`

Item.defaultProps = {
  theme: {
    fontFamily: "tahoma, arial, helvetica, sans-serif",
    fontSize: {
      sm: "11px",
    },
    toolbar: {
      button: {
        color: "#000",
        bgColor: "transparent",
        borderWidth: "1px",
        borderStyle: "solid",
        borderColor: "transparent",
        borderRadius: 0,
        hoverBgColor: "#ddefff",
        hoverBorderWidth: "1px",
        hoverBorderStyle: "solid",
        hoverBorderColor: "#99bbe8",
      },
    },
  },
}

const StyledNumberInput = styled(NumberInput)`
  min-width: 60px;
  max-width: 150px;
  margin: 0px 5px;
`

const StyledLabel = styled(Label)`
  align-items: center;
`

interface PagingToolBarProps {
  pageCount: number
  currentPage: number
  onChange: (currentPageNumber: number) => void
  onRefresh: () => void
}

export const PagingToolBar: React.FC<PagingToolBarProps> = ({
  pageCount,
  currentPage,
  onRefresh,
  onChange,
}) => {
  const changeValue = (page?: number) => {
    if (page && page >= 1 && page <= pageCount && onChange) {
      onChange(page)
    }
  }

  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = parseInt(e.target.value)
    onChange(value)
  }

  const onKeyPressed = (e: React.KeyboardEvent) => {
    if (e.key === "Enter") {
      if (
        currentPage &&
        currentPage >= 1 &&
        currentPage <= pageCount &&
        onChange
      ) {
        onChange(currentPage)
      }
    }
  }

  return (
    <Wrapper>
      <Item
        onClick={() => {
          if (currentPage !== 1) changeValue(1)
        }}
        disabled={currentPage === 1}
      >
        <FirstImage title="Первая" />
      </Item>
      <Item
        onClick={() => currentPage && changeValue(currentPage - 1)}
        disabled={currentPage === 1}
      >
        <PrevImage title="Предыдушая" />
      </Item>
      <StyledLabel>
        Стр.{" "}
        <StyledNumberInput
          value={currentPage}
          onChange={onInputChange}
          onKeyUp={onKeyPressed}
          max={pageCount}
          min={1}
        />{" "}
        из {pageCount}
      </StyledLabel>
      <Item
        onClick={() => currentPage && changeValue(currentPage + 1)}
        disabled={currentPage === pageCount}
      >
        <NextImage title="Следующая" />
      </Item>
      <Item
        onClick={() => {
          if (currentPage !== pageCount) changeValue(pageCount)
        }}
        disabled={currentPage === pageCount}
      >
        <LastImage title="Последняя" />
      </Item>
      <Item onClick={() => onRefresh()}>
        <RefreshImage title="Обновить" />
      </Item>
    </Wrapper>
  )
}
