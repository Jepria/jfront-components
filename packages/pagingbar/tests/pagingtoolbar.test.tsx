import { render } from "@testing-library/react"
import React from "react"
import { PagingToolBar } from "../src"

test("Checking for the existence of an element PagingToolBar", () => {
  render(<PagingToolBar pageCount={2} />)
})

test("Matches snapshot ", () => {
  const { asFragment } = render(<PagingToolBar pageCount={2} />)
  expect(asFragment()).toMatchSnapshot()
})
