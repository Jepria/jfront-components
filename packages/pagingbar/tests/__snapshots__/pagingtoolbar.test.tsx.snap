// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw etJfvk"
  >
    <button
      class="sc-AxhCb bjlzQy"
    >
      <img
        src="data:image/gif;base64,R0lGODlhEAAQAIcAACRIkChNli1TnjNZpjhgrz9otx1m9S9kzy9l0yxm2DBiyDFq2zhu3D5z2z913SRq8ixx7jRy6D5++URvwEJt1Ep1yE970Eh630574UD/QFOA1lOE3VWF3liH30+K9k+J/VGD4VeH4FaE6VmK4l+L4F+M4WCM5GWP62mR42iU5WuT62+W7G2Z5mec+2ma/2ic+XGa536f7oKg54Wu/42r8Iiu/5a076G+8KK9/6W//6rE7wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAABkALAAAAAAQABAAAAh6ADMIHEiwoMGDB2HIQGEwhEEWOhwSJKHBYIobFQeWyGHBoAkbHQV2yOGigsERNExm4JBjxocJBkHEgLkBR40XEgoYvLBCZ4YGLTxAIGDQgQqiAg9EeDDAIIMTTQcqMCDA4AIRVQkiCGAwAQauBQEYpABALMKzaNMODAgAOw=="
        title="Первая"
      />
    </button>
    <button
      class="sc-AxhCb bjlzQy"
    >
      <img
        src="data:image/gif;base64,R0lGODlhEAAQAIcAACRIkChNli1TnjNZpjhgrz9otx1m9S9kzy9l0yxm2DBiyDFq2zhu3D5z2z913SRq8ixx7jRy6D5++URvwEJt1Ep1yE970Eh630574UD/QFOA1lOE3VWF3liH30+K9k+J/VGD4VeH4FaE6VmK4l+L4F+M4WCM5GWP62mR42iU5WuT62+W7G2Z5mec+2ma/2ic+XGa536f7oKg54Wu/42r8Iiu/5a076G+8KK9/6W//6rE7wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAABkALAAAAAAQABAAAAhMADMIHEiwoMGDCAuGSFiQhAaGA0vksAAxQ4ccLipA5JBjxocJDDfgqPFCQgGIDVp4gECg4oEIDwZUzKDAgICZGRAEwJkBAM+fQA0GBAA7"
        title="Предыдушая"
      />
    </button>
    <label
      class="sc-AxhUy esMeTj"
    >
      Стр. 
      <input
        class="sc-AxgMl jVJRfs"
        max="2"
        min="1"
        type="number"
        value="1"
      />
       из 2
    </label>
    <button
      class="sc-AxhCb bjlzQy"
    >
      <img
        src="data:image/gif;base64,R0lGODlhEAAQAIcAACRIkChNli1TnjNZpjhgrz9otx1m9S9kzy9l0yxm2DBiyDFq2zhu3D5z2z913SRq8ixx7jRy6D5++URvwEJt1Ep1yE970Eh630574UD/QFOA1lOE3VWF3liH30+K9k+J/VGD4VeH4FaE6VmK4l+L4F+M4WCM5GWP62mR42iU5WuT62+W7G2Z5mec+2ma/2ic+XGa536f7oKg54Wu/42r8Iiu/5a076G+8KK9/6W//6rE7wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAABkALAAAAAAQABAAAAhIADMIHEiwoMGDCDPASGiQBQmGBFPgeAgxgwkcNShUHGHjRQuNDEHQaAEhwQCGF2CUBADRAYgELCEyoBAT4oKaEBNU3MmzJ8GAADs="
        title="Следующая"
      />
    </button>
    <button
      class="sc-AxhCb bjlzQy"
    >
      <img
        src="data:image/gif;base64,R0lGODlhEAAQAIcAACRIkChNli1TnjNZpjhgrz9otx1m9S9kzy9l0yxm2DBiyDFq2zhu3D5z2z913SRq8ixx7jRy6D5++URvwEJt1Ep1yE970Eh630574UD/QFOA1lOE3VWF3liH30+K9k+J/VGD4VeH4FaE6VmK4l+L4F+M4WCM5GWP62mR42iU5WuT62+W7G2Z5mec+2ma/2ic+XGa536f7oKg54Wu/42r8Iiu/5a076G+8KK9/6W//6rE7wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAABkALAAAAAAQABAAAAh4ADMIHEiwoMGDB2EYhCEDhUEWJAqy0BHCYAocEQemuKHBoAkcNSgMNGHDgsERNl60EJlhBI0KBkHQaAEhwYAMIGJMMHgBRk0AAi+sKGDQAYgEQAU6UEHAIAMKSQUyOHGz4IKoAheIEGAwQVcMARAWhIpVrNmzGQICADs="
        title="Последняя"
      />
    </button>
    <button
      class="sc-AxhCb bjlzQy"
    >
      <img
        src="data:image/gif;base64,R0lGODlhEAAQAIcAABBYsRhcvRhuzhR0xiBxzyZ4zyd70yd71SR72Cp50Dd3yTp/zT16yEN5w0J8x1J+whqI3BqA7ByQ+CKA1iqA0iyE0SqC2jyK1iyL5SqV/CqX/zSL4jqh/z2n/0qHyEqMzkqH1UqG20+J1kKQ30qS1liOzVaR3VSb10Sa4Eyf6VKf40uh4Uii60On/0it/0+u/1Kj4Vqk6lew/1my/2Wb3Wuc1W+c2nCNx26n3G6q33Sj3GCm7G+q5Guz72O3/3ax53a3+HS+/3W+/3W//3nB/37D/4Wl1oOn25Sx3YWp4Iq1442y4Y+66JC+552+5ojC74XC/YjK/47J/5TM/5bM/5jP/5vM/p7P/5/P/6XA5qHK763C5KzE6arM7KDT/6LR/6HU/6TS/6DZ/7TM6rDU87Df/7fc/73W87Hg/7rk/73g/7zi/7zl/8Tb9MTd9s7d88Dk/8Xi/87k99bh9Nbm9tHp/9rm9uPr+eXz/+b0/+vy/O31/Pb7//j5/P3+//7+/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAAIAALAAAAAAQABAAAAiuAAEJHEiwoMEmKCZQ+DDG4MAeEjJwaDHkgBOHWiA8IfMDgwwiCeaYCEGQR56BbjYIiQLAigKHA5cAmVKlTAOYgPQYWeDlC5oHArusgMEihQo5NcSwgbMmzQ2BOPDEMQOmSI43NESA8OBgi8ATdcJgqaDEYJ+BJNRcobKDCU5AF6AMkBLEQpuBd5IQHBGDTgEfMxDo4HIkQBaHZwy8cNFBQwQbOO2UICCAAZK3mAMCADs="
        title="Обновить"
      />
    </button>
  </div>
</DocumentFragment>
`;
